File: 1861-rotating-the-box/1861-rotating-the-box.py
class Solution:
    def rotateTheBox(self, box: List[List[str]]) -> List[List[str]]:
        m, n = len(box), len(box[0])
        
        for row in box:
            r = w = n - 1             
            while r >= 0:               
                if row[r] == '*':
                    w = r - 1 
                elif row[r] == '#':
                    if r < w:
                        row[r] = '.'
                        row[w] = '#'                        
                    w -= 1
                r -= 1

        return [list(col)[::-1] for col in zip(*box)]

File: 1861-rotating-the-box/1861-rotating-the-box.cpp
class Solution {
public:
    vector<vector<char>> rotateTheBox(vector<vector<char>>& box) {
        int m = box.size();
        int n = box[0].size();
        vector<vector<char>> res(n, vector<char>(m));
        
        for(int i=0; i<n; i++) {
            for(int j =0; j<m; j++) {
                res[i][j] = box[j][i];
            }
        }
        
        for(vector<char>& row:res) {
            reverse(begin(row), end(row));
        }
        
        for(int j=0; j<m; j++) {
            for(int i=n-1; i>=0; i--) {
                if(res[i][j] == '.') {
                    int stone = -1;
                    
                    for(int k = i-1; k>=0; k--) {
                        if(res[k][j] == '*') break;
                        else if(res[k][j] == '#'){
                            stone = k;
                            break;
                        }
                    }
                    
                    if(stone != -1) {
                        res[i][j] = '#';
                        res[stone][j] = '.';
                    }
                }
            }
        }
        
        
        
        return res;
    }
};